lib/AnalysisStructured/PtrAnalysis.cpp

```diff
@@ -225,13 +225,33 @@ LogicalResult PtrState::addState(const PtrState &lhsState,
       strides.push_back(builder.getIndexAttr(1));
       // New offset is offset * stride.
       auto newLhsOffset = lhsState.offsets[i];
-      if (!hasConstZero(lhsState.strides[i])) {
+      if (lhsState.hasModulo()) {
+        auto mod = expandOFRIndex(lhsState.shape[i], lhsState.offsets[i], loc,
+                                  builder);
+        // Apply modulo to the offset.
```

Comment:
i know this code was already here. but looking at this again, it's also very hard to understand why we have to expandOFRIndex. can you please document this whole section of the code again?